tools/compile_tf_graph.py

```diff
@@ -1476,13 +1480,15 @@ def main(argv):
   argparser.add_argument("--output_file", help='allowed extensions: pb, pbtxt, meta, metatxt, logdir')
   argparser.add_argument("--output_file_model_params_list", help="line-based, names of model params")
   argparser.add_argument("--output_file_state_vars_list", help="line-based, name of state vars")
+  argparser.add_argument("--update_i_in_graph", action="store_true", help="whether to update i in the graph")
```

Comment:
You are changing the logic from before here. Before, it was always True. Now it is False by default and can optionally be set to True.

I wonder if the default should be as before.

Reply:
Although, the default maybe should match the RASR logic. The question is more if we want to cleanup and generalize RASR.

Reply:
Maybe this changed default is ok. I assume this is also how it was in @ZhouW321's fork.

Reply:
Btw, there is also the `cond` global state var. I just noticed that we currently exclude this for some strange reason. Which is maybe again for RASR because RASR currently has this strange assumption that there is only exactly one single global state var which is `i`.

Reply:
Any comment?

Reply:
Ah, sorry. I think because RASR currently has this assumption, it might be good if the default behavior of the compile script would match this assumption